emar.py

```diff
@@ -73,15 +67,12 @@ def run():
         parts[0] += '_' + h
         newname = '.'.join(parts)
         full_newname = os.path.join(dirname, newname)
-        assert not os.path.exists(full_newname)
+        newargs[j] = full_newname
         try:
           shutil.copyfile(orig_name, full_newname)
-          newargs[j] = full_newname
           to_delete.append(full_newname)
-          contents.add(newname)
         except Exception:
           # it is ok to fail here, we just don't get hashing
-          contents.add(basename)
```

Comment:
after removing this line, is the comment above where it was still correct?

it seems like hashing and duplicate names are two separate issues. this PR makes it assume the hashing part always works, and repurposes the try-except from one purpose to another. are we not worried about catching hashing failures any more?

Reply:
Ah yes, I move the newargs[j] assignment back into the try block so this comment could be meaningful again.